### PR TITLE
issue-46 to web-dev

### PR DESCRIPTION
**Pull request by sorianog**
Created the GET endpoint for retrieving state regulations data
Added two tests for the GET endpoints

### CODE REVIEW

File: web/app/routes/mainRoutes.js [hunk 1]
```diff
@@ -115,4 +116,14 @@ router.delete("/drivingschools/:schoolId/instructors/:userId", function(req, res
 	res.json({});
 });
 
+// GET the driving regulations for a specified state
+rouer.get("/stateregulations/:state", function(req, res) {
```
Comment (zawadzkip) [inline web/app/routes/mainRoutes.js]: This won't work, `rouer` is misspelled

File: web/app/controllers/stateRegsCtrl.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+var StateRegulations = require("../models/StateRegulations");
+
+exports.getStateRegs = function (state, callback, error) {
+	StateRegulations.findOne({
+		state: state
```
Comment (zawadzkip) [inline web/app/controllers/stateRegsCtrl.js:5]: is this an id? What is `state`?

File: web/app/routes/mainRoutes.js [hunk 3]
```diff
@@ -115,4 +116,14 @@ router.delete("/drivingschools/:schoolId/instructors/:userId", function(req, res
...
+	stateRegsCtrl.getStateRegs(req.params.state, (sateRegs)=>{
```
Comment (zawadzkip) [inline web/app/routes/mainRoutes.js]: `sateregs` is not spelled correctly, this won't run because `stateregs` below is undefined

**Comment by zawadzkip** [comment]
@sorianog please review the comments, I think there is somethings messed up with your linter, the code shouldn't pass linting.

**Review by zawadzkip — DISMISSED** [review]

File: web/app/routes/mainRoutes.js (hunk 3 above)
Comment (Awkewainze) [inline web/app/routes/mainRoutes.js]: sateRegs -> stateRegs

File: web/app/routes/mainRoutes.js [hunk 4]
```diff
@@ -115,4 +116,14 @@ router.delete("/drivingschools/:schoolId/instructors/:userId", function(req, res
...
+		res.json(stateRegs);
+	}, (error)=>{
```
Comment (Awkewainze) [inline web/app/routes/mainRoutes.js]: error -> err

File: web/app/routes/mainRoutes.js (hunk 1 above)
Comment (Awkewainze) [inline web/app/routes/mainRoutes.js]: rouer -> router

File: web/app/controllers/stateRegsCtrl.js (hunk 2 above)
Comment (sorianog) [reply web/app/controllers/stateRegsCtrl.js:5]: State is the parameter that gets passed in to the request when trying to get state regulations.

File: web/app/routes/mainRoutes.js (hunk 1 above)
Comment (sorianog) [reply web/app/routes/mainRoutes.js]: Yes, `router` is the correct way to spell it! Mahalo.

Comment (sorianog) [reply web/app/routes/mainRoutes.js]: fixed

File: web/app/routes/mainRoutes.js (hunk 3 above)
Comment (sorianog) [reply web/app/routes/mainRoutes.js]: fixed. now `stateRegs`

File: web/app/routes/mainRoutes.js (hunk 4 above)
Comment (sorianog) [reply web/app/routes/mainRoutes.js]: fixed. now `err`

File: web/app/controllers/stateRegsCtrl.js (hunk 2 above)
Comment (zawadzkip) [reply web/app/controllers/stateRegsCtrl.js:5]: can we add a check for string casing and whitespace? any extra whitespace and upper/lowercase might cause this to fail a query

**Review by dobber1611 — APPROVED** [review]

**Pull request merged**